From b5bcdd7023adcac8cb78d3c75534e4d150184fd4 Mon Sep 17 00:00:00 2001
From: robinz1x <robinx.zhang@intel.com>
Date: Thu, 19 Jul 2018 14:48:26 +0800
Subject: [PATCH] Check whether guest user is already created or not

UserManagerHelper and SystemUI part:
SystemUI UserGridRecyclerView is using UserManagerHelper, implement
findCurrentGuestUser() api in UserManagerHelper, and check whether
the guest user is created when click "Guest" icon.

If guest user already created, switch to it;
otherwise, create a new guest user.

Change-Id: Ieca381228abca4da4374a31b29229c15eb23a55e
Tracked-On: OAM-72293
Signed-off-by: robinz1x <robinx.zhang@intel.com>
Signed-off-by: Yan, WalterX <walterx.yan@intel.com>
---
 .../com/android/settingslib/users/UserManagerHelper.java  | 15 +++++++++++++++
 .../systemui/statusbar/car/UserGridRecyclerView.java      |  8 +++++++-
 2 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/packages/SettingsLib/src/com/android/settingslib/users/UserManagerHelper.java b/packages/SettingsLib/src/com/android/settingslib/users/UserManagerHelper.java
index 4c45a75..31bfbda 100644
--- a/packages/SettingsLib/src/com/android/settingslib/users/UserManagerHelper.java
+++ b/packages/SettingsLib/src/com/android/settingslib/users/UserManagerHelper.java
@@ -392,6 +392,21 @@ public final class UserManagerHelper {
     }
 
     /**
+     * Find the current guest user.
+     */
+    public UserInfo findCurrentGuestUser() {
+        final List<UserInfo> users = getAllUsers();
+        final int size = users.size();
+        for (int i = 0; i < size; i++) {
+            final UserInfo user = users.get(i);
+            if (user.isGuest() && !user.guestToRemove) {
+                return user;
+            }
+        }
+        return null;
+    }
+
+    /**
      * Gets an icon for the user.
      *
      * @param userInfo User for which we want to get the icon.
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
index a49d507..5ca8183 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
@@ -214,7 +214,13 @@ public class UserGridRecyclerView extends PagedListView implements
                 // If the user selects Guest, start the guest session.
                 if (userRecord.mIsStartGuestSession) {
                     notifyUserSelected(userRecord);
-                    mUserManagerHelper.startNewGuestSession(mGuestName);
+                    UserInfo currentGuestUser = mUserManagerHelper.findCurrentGuestUser();
+                    // If guest user already created, switch to it; otherwise, create a new guest user
+                    if (currentGuestUser != null) {
+                        mUserManagerHelper.switchToUser(currentGuestUser);
+                    } else {
+                        mUserManagerHelper.startNewGuestSession(mGuestName);
+                    }
                     return;
                 }
 
-- 
1.9.1

